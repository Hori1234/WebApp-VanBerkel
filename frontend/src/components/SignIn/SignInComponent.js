import React, { Component } from "react";
import {
  Layout,
  message,
  Avatar,
  Form,
  Input,
  Button,
  Checkbox,
  Typography,
  Divider,
} from "antd";
import "antd/dist/antd.css";
import {
  UserOutlined
} from "@ant-design/icons";

const axios = require("axios");
const { Title } = Typography;

export default class SignInComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      role: "tmp",
      username: "",
      password: "",
      valid: false,
    };
    this.handleChangeUsername = this.handleChangeUsername.bind(this);
    this.handleChangePassword = this.handleChangePassword.bind(this);
    this.onFinish = this.onFinish.bind(this);
  }

  postCredentials = async (uss, pass) => {
    var responseRole = "";
    var isValidated = false;
    console.log(uss, pass);
    await axios
      .post("/api/auth/login", {
        username: uss,
        password: pass,
        remember: true,
      })
      .then(function (response) {
        console.log(response);
        console.log(response.data.role);
        //this.props.userRole = response.data.role;
        responseRole = response.data.role;
        isValidated = true;
        console.log("===============");
      })
      .catch(function (error) {
        console.log(error);
        isValidated = false;
      });

    console.log("waiting for post response");
    //responseRole = "view only";
    this.onChangeRole(responseRole, isValidated);
    console.log(this.state.role);
  };

  onChangeRole = (value, validated) => {
    this.setState({
      role: value,
      valid: validated,
    });
  };

  onAuthentificate = async () => {
    var uss = this.getUsername();
    var pass = this.getPassword();
    console.log(uss);
    await this.postCredentials(uss, pass);

    if (this.state.valid) {
      console.log("check state: " + this.state.role);
      this.props.changeUser(this.state.role);
      this.props.changeState();
    } else {
      message.info("accont not valid");
    }
  };
  getUsername = () => {
    return this.state.username;
  };
  getPassword = () => {
    return this.state.password;
  };
  handleChangeUsername = (event) => {
    this.setState({
      username: event.target.value,
    });
    console.log(this.state.username);
  };
  handleChangePassword = (event) => {
    this.setState({
      password: event.target.value,
    });
    console.log(this.state.password);
  };

  onFinish = (event) => {
    console.log(
      "Ussername and Password set",
      this.state.username,
      this.state.password
    );
  };
  onFinishFailed = (errorInfo) => {
    console.log("Failed:", errorInfo);
    message.info("fill all the available spaces");
  };

  render() {
    const layout = {
      labelCol: {
        span: 8,
      },
      wrapperCol: {
        span: 16,
      },
    };
    const tailLayout = {
      wrapperCol: {
        offset: 8,
        span: 16,
      },
    };

    return (
      <Layout style={{ margin: 30, height: "95vh", display: "flex" }}>
        <Layout style={{ alignItems: "center", marginTop: 15 }}>
          <Avatar
            style={{ marginTop: 100 }}
            shape="square"
            size={64}
            icon={<UserOutlined />}
          />
          <Divider style={{ marginTop: 10 }}>Welcome</Divider>
        </Layout>

        <Layout
          style={{
            alignItems: "center",
            display: "flex",
            marginTop: -70,
            marginBottom: -120,
          }}
        >
          <Title style={{ marginLeft: -280 }} level={5.2}>
            Sign In
          </Title>
          <Title style={{ marginLeft: -190 }} level={5}>
            Fil the form in order to login
          </Title>
        </Layout>

        <Layout
          style={{ alignItems: "center", display: "flex", marginTop: -50 }}
        >
          <Form
            {...layout}
            name="basic"
            initialValues={{ remember: true }}
            onFinish={this.onFinis}
            onFinishFailed={this.onFinishFailed}
          >
            <Form.Item
              label="Username"
              name="username"
              rules={[
                { required: true, message: "Please input your username!" },
              ]}
              style={{ width: "100%", marginRight: 180, marginLeft: -50 }}
            >
              <Input
                value={this.state.username}
                onChange={this.handleChangeUsername}
              />
            </Form.Item>

            <Form.Item
              label="Password"
              name="password"
              rules={[
                { required: true, message: "Please input your password!" },
              ]}
              style={{ width: "100%", marginRight: 180, marginLeft: -50 }}
            >
              <Input.Password
                value={this.state.password}
                onChange={this.handleChangePassword}
              />
            </Form.Item>

            <Form.Item
              {...tailLayout}
              name="remember"
              valuePropName="checked"
              style={{ width: "100%", marginRight: 180, marginLeft: -50 }}
            >
              <Checkbox>Remember me</Checkbox>
            </Form.Item>

            <Form.Item {...tailLayout}>
              <Button
                type="primary"
                htmlType="submit"
                style={{ marginLeft: 10, marginTop: 60, width: "60%" }}
                onClick={this.onAuthentificate.bind(this)}
              >
                Login
              </Button>
            </Form.Item>
          </Form>
        </Layout>
      </Layout>
    );
  }
}
