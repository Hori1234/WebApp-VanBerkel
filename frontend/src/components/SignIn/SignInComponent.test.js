import React from "react";
import SignInComponent from "./SignInComponent";
import {shallow} from "enzyme";

describe("SignInComponent", () => {
  const component = shallow(<SignInComponent />);

  it("should render the login page correctly", () => {
    expect(component).toMatchSnapshot();
  });

  const roles = ["administrator", "view-only", "planner"];

  roles.forEach((role) => {
    it("should set the correct role", () => {
      component.setState({ role });
      expect(component.instance().state.role).toEqual(role);
    });
  });

  const usernames = ["bobbyz", "andrewossuna", "johnlopez"];

  usernames.forEach((username) => {
    it("should set the correct userName", () => {
      component.setState({ username });
      expect(component.instance().state.username).toEqual(username);
    });
  });

  it("should set the correct password", () => {
    const password = "notapassword";
    component.setState({ password });

    expect(component.instance().state.password).toEqual(password);
  });

  const passwords = ["3284328!%$", "1093ann", "notapassword"];

  passwords.forEach((password) => {
    it("should set the correct password", () => {
      component.setState({ password });
      expect(component.instance().state.password).toEqual(password);
    });
  });
});
