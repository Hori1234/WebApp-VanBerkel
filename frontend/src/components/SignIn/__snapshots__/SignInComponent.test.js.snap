// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInComponent should render the login page correctly 1`] = `
<Layout
  style={
    Object {
      "display": "flex",
      "height": "95vh",
      "margin": 30,
    }
  }
>
  <Layout
    style={
      Object {
        "alignItems": "center",
        "marginTop": 15,
      }
    }
  >
    <Avatar
      icon={<ForwardRef(UserOutlined) />}
      shape="square"
      size={64}
      style={
        Object {
          "marginTop": 100,
        }
      }
    />
    <Divider
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      Welcome
    </Divider>
  </Layout>
  <Layout
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "marginBottom": -120,
        "marginTop": -70,
      }
    }
  >
    <Title
      level={5.2}
      style={
        Object {
          "marginLeft": -280,
        }
      }
    >
      Sign In
    </Title>
    <Title
      level={5}
      style={
        Object {
          "marginLeft": -190,
        }
      }
    >
      Fil the form in order to login
    </Title>
  </Layout>
  <Layout
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "marginTop": -50,
      }
    }
  >
    <ForwardRef(InternalForm)
      initialValues={
        Object {
          "remember": true,
        }
      }
      labelCol={
        Object {
          "span": 8,
        }
      }
      name="basic"
      onFinishFailed={[Function]}
      wrapperCol={
        Object {
          "span": 16,
        }
      }
    >
      <FormItem
        label="Username"
        name="username"
        rules={
          Array [
            Object {
              "message": "Please input your username!",
              "required": true,
            },
          ]
        }
        style={
          Object {
            "marginLeft": -50,
            "marginRight": 180,
            "width": "100%",
          }
        }
      >
        <Input
          onChange={[Function]}
          type="text"
          value=""
        />
      </FormItem>
      <FormItem
        label="Password"
        name="password"
        rules={
          Array [
            Object {
              "message": "Please input your password!",
              "required": true,
            },
          ]
        }
        style={
          Object {
            "marginLeft": -50,
            "marginRight": 180,
            "width": "100%",
          }
        }
      >
        <Password
          action="click"
          iconRender={[Function]}
          onChange={[Function]}
          value=""
          visibilityToggle={true}
        />
      </FormItem>
      <FormItem
        name="remember"
        style={
          Object {
            "marginLeft": -50,
            "marginRight": 180,
            "width": "100%",
          }
        }
        valuePropName="checked"
        wrapperCol={
          Object {
            "offset": 8,
            "span": 16,
          }
        }
      >
        <Checkbox
          indeterminate={false}
        >
          Remember me
        </Checkbox>
      </FormItem>
      <FormItem
        wrapperCol={
          Object {
            "offset": 8,
            "span": 16,
          }
        }
      >
        <Button
          block={false}
          ghost={false}
          htmlType="submit"
          loading={false}
          onClick={[Function]}
          style={
            Object {
              "marginLeft": 10,
              "marginTop": 60,
              "width": "60%",
            }
          }
          type="primary"
        >
          Login
        </Button>
      </FormItem>
    </ForwardRef(InternalForm)>
  </Layout>
</Layout>
`;
